chainer/functions/split_axis.py

```diff
@@ -26,6 +29,50 @@ def __init__(self, indices_or_sections, axis):
         self.indices_or_sections = indices_or_sections
         self.axis = axis
 
+    def check_type_forward(self, in_types):
+        type_check.expect(
+            in_types.size() == 1,
+            in_types[0].ndim >= self.axis
+        )
+
+        if isinstance(self.indices_or_sections, collections.Iterable):
+            max_index = type_check.Variable(
+                self.indices_or_sections[-1], 'max_index')
+            type_check.expect(in_types[0].shape[self.axis] >= max_index)
```

Comment:
I made a mistake. If the equality holds, last tensor of output is empty, which is forbidden.